### PR TITLE
Add Profiler table kwargs

### LINKED ISSUE #17367: Expose more arguments for PyTorchProfiler
### Description & Motivation

There are only limited arguments for PyTorch profiler exposed to the user. Some other arguments might be useful, such as `max_name_column_width` when the path is long and truncated. 

### Pitch

Expose more arguments of PyTorch profiler such as `max_name_column_width`. 

### Alternatives

Customize a lightning profiler to get a new PyTorch profiler. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by baskrahmer**
## What does this PR do?

Adds table_kwargs to the PyTorchProfiler class to expose more formatting options to the user directly.

Fixes #17367

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: tests/tests_pytorch/profilers/test_profiler.py [hunk 1]
```diff
@@ -619,3 +620,37 @@ def test_profile_callbacks(tmpdir):
         e.name == "[pl][profile][Callback]EarlyStopping{'monitor': 'train_loss', 'mode': 'min'}.on_validation_start"
         for e in pytorch_profiler.function_events
     )
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="requires Python 3.10 or higher")
```
Comment (carmocca) [inline tests/tests_pytorch/profilers/test_profiler.py:625]:
```suggestion
@RunIf(min_python="3.10")
```

File: tests/tests_pytorch/profilers/test_profiler.py [hunk 2]
```diff
@@ -619,3 +620,37 @@ def test_profile_callbacks(tmpdir):
...
+def test_profiler_table_kwargs_summary_length(tmpdir):
+    """Test if setting max_name_column_width in table_kwargs changes summary length."""
+
+    summaries = []
+    # Default table_kwargs (None) sets max_name_column_width to 55
+    for table_kwargs in [None, {"max_name_column_width": 20}, {"max_name_column_width": 100}]:
+        pytorch_profiler = PyTorchProfiler(dirpath=tmpdir, filename="profile", table_kwargs=table_kwargs)
+        model = BoringModel()
+        trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, fast_dev_run=1, profiler=pytorch_profiler)
+        trainer.fit(model)
```
Comment (carmocca) [inline tests/tests_pytorch/profilers/test_profiler.py:635]: Can you add data to the profile manually instead of starting up a Trainer just to get some data in?

File: src/lightning/pytorch/profilers/pytorch.py [hunk 3]
```diff
@@ -314,6 +317,19 @@ def __init__(
...
+                raise KeyError(
+                    f"Found invalid table_kwargs key: {key}. This is already a positional argument of the Profiler."
+                )
+            valid_table_keys = set(inspect.signature(EventList.table).parameters.keys()) - {
+                "self",
+                "sort_by",
+                "row_limit",
+            }
+            if key not in valid_table_keys:
+                raise KeyError(f"Found invalid table_kwargs key: {key}. Should be within {valid_table_keys}.")
```
Comment (carmocca) [inline src/lightning/pytorch/profilers/pytorch.py:331]: This validation is a bit overkill. I don't think we do this anywhere else we pass kwargs

**Review by carmocca — COMMENTED** [review]
Looks great

File: src/lightning/pytorch/profilers/pytorch.py (hunk 3 above)
Comment (baskrahmer) [reply src/lightning/pytorch/profilers/pytorch.py:331]: Yes in retrospect this probably isn't necessary. The reason I did this though is to make it fail immediately as opposed to when the table summary is created.

Shall I change it to this structure? I.e. just let it run with valid kwargs.
```
    def _create_profiler(self, profiler: Type[_PROFILER]) -> _PROFILER:
        init_parameters = inspect.signature(profiler.__init__).parameters
        kwargs = {k: v for k, v in self._profiler_kwargs.items() if k in init_parameters}
        return profiler(**kwargs)
```

Comment (carmocca) [reply src/lightning/pytorch/profilers/pytorch.py:331]: It's okay. Let's leave it as it is.

The profiler logic is because there's different profiler classes

**Review by carmocca — APPROVED** [review]
Good job!
